### PR TITLE
updated upgrade script to retry slower

### LINKED ISSUE #5: Upgrade script GET retry is too fast
#### What is the current behavior?

If the upgrade script GET fails, we retry up to N times thanks to the usual controller requeue mechanism. However, we endup retrying way too fast, which will not help in the case of a transient issue.

#### What is the expected behavior?

Insert a delay between GET retries.

#### Additional Context

The `pester` library which we're considering integrating may also help with this issue. TODO ticket incoming.

### Reproduction
1. Launch a cluster using `csutil` locally
2. Force an issue with provision service, or change the provision base URL, etc.
3. See retry too fast

### Evidence

```
2018-10-24T18:52:15.684Z        INFO    Downloading upgrade script
2018-10-24T18:52:25.578Z        DEBUG   Request: "GET /v3/organizations/5813c2b6-9bf2-43ba-bc60-f3711e4e4d8a/clusters/6f6e4c6d-ee67-45ec-a5c2-6115db115eb5/nodes/20f75517-34f6-40ed-8eb6-21bb62e7bb64/upgrade/kubernetes?version=1.12.1 HTTP/1.1\r\nHost: stage-provision.containership.io\r\nUser-Agent: Go-http-client/1.1\r\nAuthorization: JWT eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJjbHVzdGVyX2lkIjoiNmY2ZTRjNmQtZWU2Ny00NWVjLWE1YzItNjExNWRiMTE1ZWI1IiwibWV0YWRhdGEiOnsidHlwZSI6ImNsdXN0ZXIifSwiaWF0IjoxNTQwNDA1NDAwLCJleHAiOjE1NzE5NDE0MDAsImlzcyI6ImNvbnRhaW5lcnNoaXAuY2xvdWQuYXV0aCJ9.6KzGp5_y3YQFuah2i9vw3ifKi67ktpL7grZLekZG9HS0Rkp-QN_zIN-O20scgBS-g-I61_SfFTrLb8iLBN11wA\r\nAccept-Encoding: gzip\r\n\r\n"
2018-10-24T18:52:25.578Z        ERROR   Download upgrade script failed:Get https://stage-provision.containership.io/v3/organizations/5813c2b6-9bf2-43ba-bc60-f3711e4e4d8a/clusters/6f6e4c6d-ee67-45ec-a5c2-6115db115eb5/nodes/20f75517-34f6-40ed-8eb6-21bb62e7bb64/upgrade/kubernetes?version=1.12.1: dial tcp: lookup stage-provision.containership.io on 10.96.0.10:53: read udp 192.168.0.3:38924->10.96.0.10:53: i/o timeout
github.com/containership/cloud-agent/pkg/log.Error
        /go/src/github.com/containership/cloud-agent/pkg/log/log.go:94
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).startUpgrade
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:259
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).syncHandler
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:242
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).processNextWorkItem.func1
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:140
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).processNextWorkItem
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:142
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).runWorker
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:114
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).(github.com/containership/cloud-agent/pkg/agent.runWorker)-fm
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:102
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
2018-10-24T18:52:25.578Z        ERROR   error syncing 'containership-core/91d2f140-7f81-4031-a91f-851bc3eafd04': Get https://stage-provision.containership.io/v3/organizations/5813c2b6-9bf2-43ba-bc60-f3711e4e4d8a/clusters/6f6e4c6d-ee67-45ec-a5c2-6115db115eb5/nodes/20f75517-34f6-40ed-8eb6-21bb62e7bb64/upgrade/kubernetes?version=1.12.1: dial tcp: lookup stage-provision.containership.io on 10.96.0.10:53: read udp 192.168.0.3:38924->10.96.0.10:53: i/o timeout. Has been resynced 5 times
github.com/containership/cloud-agent/pkg/log.Error
        /go/src/github.com/containership/cloud-agent/pkg/log/log.go:94
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).processNextWorkItem
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:145
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).runWorker
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:114
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).(github.com/containership/cloud-agent/pkg/agent.runWorker)-fm
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:102
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
2018-10-24T18:52:25.658Z        DEBUG   UpgradeAgentController: processing key="containership-core/91d2f140-7f81-4031-a91f-851bc3eafd04"
2018-10-24T18:52:25.659Z        INFO    Beginning upgrade process
2018-10-24T18:52:25.659Z        INFO    Downloading upgrade script
2018-10-24T18:52:35.665Z        DEBUG   Request: "GET /v3/organizations/5813c2b6-9bf2-43ba-bc60-f3711e4e4d8a/clusters/6f6e4c6d-ee67-45ec-a5c2-6115db115eb5/nodes/20f75517-34f6-40ed-8eb6-21bb62e7bb64/upgrade/kubernetes?version=1.12.1 HTTP/1.1\r\nHost: stage-provision.containership.io\r\nUser-Agent: Go-http-client/1.1\r\nAuthorization: JWT eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJjbHVzdGVyX2lkIjoiNmY2ZTRjNmQtZWU2Ny00NWVjLWE1YzItNjExNWRiMTE1ZWI1IiwibWV0YWRhdGEiOnsidHlwZSI6ImNsdXN0ZXIifSwiaWF0IjoxNTQwNDA1NDAwLCJleHAiOjE1NzE5NDE0MDAsImlzcyI6ImNvbnRhaW5lcnNoaXAuY2xvdWQuYXV0aCJ9.6KzGp5_y3YQFuah2i9vw3ifKi67ktpL7grZLekZG9HS0Rkp-QN_zIN-O20scgBS-g-I61_SfFTrLb8iLBN11wA\r\nAccept-Encoding: gzip\r\n\r\n"
2018-10-24T18:52:35.665Z        ERROR   Download upgrade script failed:Get https://stage-provision.containership.io/v3/organizations/5813c2b6-9bf2-43ba-bc60-f3711e4e4d8a/clusters/6f6e4c6d-ee67-45ec-a5c2-6115db115eb5/nodes/20f75517-34f6-40ed-8eb6-21bb62e7bb64/upgrade/kubernetes?version=1.12.1: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
github.com/containership/cloud-agent/pkg/log.Error
        /go/src/github.com/containership/cloud-agent/pkg/log/log.go:94
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).startUpgrade
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:259
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).syncHandler
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:242
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).processNextWorkItem.func1
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:140
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).processNextWorkItem
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:142
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).runWorker
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:114
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).(github.com/containership/cloud-agent/pkg/agent.runWorker)-fm
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:102
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
2018-10-24T18:52:35.665Z        INFO    Dropping containership-core/91d2f140-7f81-4031-a91f-851bc3eafd04 out of the queue: Get https://stage-provision.containership.io/v3/organizations/5813c2b6-9bf2-43ba-bc60-f3711e4e4d8a/clusters/6f6e4c6d-ee67-45ec-a5c2-6115db115eb5/nodes/20f75517-34f6-40ed-8eb6-21bb62e7bb64/upgrade/kubernetes?version=1.12.1: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
2018-10-24T18:52:35.665Z        ERROR   Get https://stage-provision.containership.io/v3/organizations/5813c2b6-9bf2-43ba-bc60-f3711e4e4d8a/clusters/6f6e4c6d-ee67-45ec-a5c2-6115db115eb5/nodes/20f75517-34f6-40ed-8eb6-21bb62e7bb64/upgrade/kubernetes?version=1.12.1: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
github.com/containership/cloud-agent/pkg/log.Error
        /go/src/github.com/containership/cloud-agent/pkg/log/log.go:94
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).processNextWorkItem
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:145
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).runWorker
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:114
github.com/containership/cloud-agent/pkg/agent.(*UpgradeController).(github.com/containership/cloud-agent/pkg/agent.runWorker)-fm
        /go/src/github.com/containership/cloud-agent/pkg/agent/upgrade_controller.go:102
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/containership/cloud-agent/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
2018-10-24T19:02:05.319Z        DEBUG   UpgradeAgentController: processing key="containership-core/91d2f140-7f81-4031-a91f-851bc3eafd04"
2018-10-24T19:02:05.319Z        DEBUG   Upgrade file "current" is being removed.
```

### PR DESCRIPTION
**Pull request by ashleyschuett**
### Description

Changes the exponential retry to a second vs 5 milliseconds, this makes to retry take longer. Currently, this puts 30 seconds between the first and the last request, we can, of course, increase this. Since we have a node timeout of 10 (i think?) minutes I didn't want it to be too long but would be fine changing it to be a little longer if we think that would be better. 

 #### What does this pull request accomplish?

 #### What issue(s) does this fix?
* Fixes #5 

 #### Additional Considerations

 ### Testing

`make build-agent`
`docker tag containership/cloud-agent:latest ashleyschuett/cloud-agent:latest`
`docker push ashleyschuett/cloud-agent:latest`

- changed cloud-agent plugin to use ashleyschuett/cloud-agent:latest

```
$ kubectl -n containership-core get ds cloud-agent -o json | jq '.spec.template.spec.containers[].image'
"ashleyschuett/cloud-agent:latest"
```
```
$ kubectl -n containership-core logs cloud-agent-b9p9n -f | grep "Downloading upgrade script"
2019-02-14T19:05:31.829Z	INFO	Downloading upgrade script
2019-02-14T19:05:33.834Z	INFO	Downloading upgrade script
2019-02-14T19:05:37.840Z	INFO	Downloading upgrade script
2019-02-14T19:05:45.845Z	INFO	Downloading upgrade script
2019-02-14T19:06:01.856Z	INFO	Downloading upgrade script
```

 #### Setup

 #### Instructions

 ### Dependencies

### CODE REVIEW

File: pkg/agent/upgrade_controller.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 	"io/ioutil"
 	"time"
 
+	"golang.org/x/time/rate"
```
Comment (mattkelly) [inline pkg/agent/upgrade_controller.go]: Split off from `k8s.io` imports

Comment (ashleyschuett) [reply pkg/agent/upgrade_controller.go]: it's the same package that the client-go workqueue package is using https://github.com/kubernetes/client-go/blob/v9.0.0/util/workqueue/default_rate_limiters.go#L24

Comment (mattkelly) [reply pkg/agent/upgrade_controller.go]: I'm just saying there should be a newline between this and the k8s.io imports

Comment (ashleyschuett) [reply pkg/agent/upgrade_controller.go]: ooo.. that makes sense. 👍 my bad

**Pull request merged**